spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpPropertiesConfigAdapter.java

```diff
@@ -43,6 +43,11 @@ class OtlpPropertiesConfigAdapter extends StepRegistryPropertiesConfigAdapter<Ot
 	 */
 	private static final String DEFAULT_APPLICATION_NAME = "unknown_service";
 
+	/**
+	 * Default value for application group if {@code spring.application.group} is not set.
+	 */
+	private static final String DEFAULT_APPLICATION_GROUP = "unknown_group";
```

Comment:
I think `default` is better than `unknown` here.

Reply:
I disagree. `unknown_group` aligns with the existing `unknown_service` that's used when there's no application name.

Reply:
I'm afraid I still disagree. This feature isn't specific to Kubernetes so aligning with its default namespace does not make sense. Furthermore, it isn't a default group but an indication that we do not know the application's group. As such, I think `unknown_group` is more meaningful as it accurately reflects the situation that we're in.

Reply:
The service name is supposed to be unique per application. It not being specified means that it is indeed unknown, as there cannot be any sensible default value that actually makes sense. The groups are shared, so it not being specified is a pretty good indicator that it belongs to the (shared) default group, not the shared unknown group. `unknown_group` makes it seem like there is something wrong with the application configuration, whereas you have just opted to not specify an actual group.

Reply:
>  It not being specified means that it is indeed unknown, as there cannot be any sensible default value that actually makes sense.

That is the case here.

> The groups are shared, so it not being specified is a pretty good indicator that it belongs to the (shared) default group

There's no "the groups" here. As described in #39913, a group can be a company, org, or team, or pretty much whatever else makes sense for grouping a set of applications. There's no concept of a default group as we don't know that such a thing makes sense in any given deployment environment.

> `unknown_group` makes it seem like there is something wrong with the application configuration, whereas you have just opted to not specify an actual group.

"unknown" doesn't imply that something is wrong, all it means is that the group is unknown. I think that accurately reflects the situation as the group really is not known.

Reply:
IMO Tag `service.group` shouldn't be added if it's `unknown_group`, It doesn't make sense from observation perspective but increase network packet size.

Reply:
I agree. We only set `service.name` to `unknown_application` because it's a [required attribute in otel](https://opentelemetry.io/docs/specs/semconv/resource/#service). This isn't the case for `service.group`, so we can just omit it, if it's not set.